src/ui/public/agg_types/param_types/field.js

```diff
@@ -55,30 +55,13 @@ FieldParamType.prototype.serialize = function (field) {
  */
 FieldParamType.prototype.getFieldOptions = function (aggConfig) {
   const indexPattern = aggConfig.getIndexPattern();
-  let fields = indexPattern.fields.raw;
-
-  if (this.onlyAggregatable) {
-    fields = fields.filter(f => f.aggregatable);
-  }
-
-  if (!this.scriptable) {
-    fields = fields.filter(field => !field.scripted);
-  }
-
-  if (this.filterFieldTypes) {
-    let filters = this.filterFieldTypes;
-    if (_.isFunction(this.filterFieldTypes)) {
-      filters = this.filterFieldTypes.bind(this, aggConfig.vis);
-    }
-    fields = propFilter('type')(fields, filters);
-    fields = _.sortBy(fields, ['type', 'name']);
-  }
-
+  const fields = aggTypeFieldFilters
+    .filter(indexPattern.fields.raw, aggConfig.type, indexPattern, aggConfig);
 
   return new IndexedArray({
     index: ['name'],
     group: ['type'],
-    initialSet: fields
+    initialSet: _.sortBy(fields, ['type', 'name']),
```

Comment:
I know that's not introduced by you, but since that's the only line using lodash in that file, maybe switch over to named import: `import { sortBy } from 'lodash';`

Reply:
fixed!